src/__tests__/unitTesting.js

```diff
@@ -0,0 +1,323 @@
+    test('saveChart function saves chart with 3 datapoints, name, and color', () => {
+        //Arrange
+        window.localStorage.clear()  //clear local storage
+        const chartList = [
+            {
+                type: "line",
+                data: [
+                    {x: 1, y: 2},
+                    {x: 5, y: 9},
+                    {x: 6, y: 10},
+                    {x: 7, y: 11}
+                ],
+                xLabel: "Time",
+                yLabel: "Temperature",
+                title: "Summer Temperatures, 2023",
+                color: "red"
+            }
+        ];
+        const newChart = {
+            type: "bar",
+            data: [
+                {x: 2, y: 3},
+                {x: 6, y: 10},
+                {x: 7, y: 11}
+            ],
+            xLabel: "Month",
+            yLabel: "Rainfall (cm)",
+            title: "Summer Rainfall, 2023",
+            color: "blue"
+        }
+        window.localStorage.setItem("savedCharts", JSON.stringify(chartList))
+
+        //Act
+        saveChart(newChart, null)  //should be save to location 0, since no other charts exist
+        
+        //Assert
+        expect(JSON.parse(window.localStorage.getItem("savedCharts")).length).toBe(2)
+        expect(JSON.parse(window.localStorage.getItem("savedCharts"))[1]).toMatchObject(newChart)
+    })
+    test('saveChart function replaces chart at index 0 with chart with 3 datapoints, name, and color', () => {
+        //Arrange
+        window.localStorage.clear()
+        const chartList = [
+            {
+                type: "line",
+                data: [
+                    {x: 1, y: 2},
+                    {x: 5, y: 9},
+                    {x: 6, y: 10},
+                    {x: 7, y: 11}
+                ],
+                xLabel: "Time",
+                yLabel: "Temperature",
+                title: "Summer Temperatures, 2023",
+                color: "red"
+            }
+        ];
+        const newChart = {
+            type: "bar",
+            data: [
+                {x: 2, y: 3},
+                {x: 6, y: 10},
+                {x: 7, y: 11}
+            ],
+            xLabel: "Month",
+            yLabel: "Rainfall (cm)",
+            title: "Summer Rainfall, 2023",
+            color: "blue"
+        };
+        window.localStorage.setItem("savedCharts", JSON.stringify(chartList))
+        
+        //Act
+        saveChart(newChart, 0)  //should be save to location 0, since no other charts exist
+        
+        //Assert
+        expect(JSON.parse(window.localStorage.getItem("savedCharts")).length).toBe(1)
+        expect(JSON.parse(window.localStorage.getItem("savedCharts"))[0]).toMatchObject(newChart)
+    })
+
+    //loadAllSavedCharts
+    test('loadAllSavedCharts returns an empty array when no charts are saved', () => {
+        //Assert
+        window.localStorage.clear()
+        expect(loadAllSavedCharts()).toMatchObject([])  //need to make that an empty array?
```

Comment:
This is me being picky, but could you remove the outdated comments that say "need to make that an empty array?"